[components,docs] add explanation of overridden `get_spec` on `Component`

cmpadden opened the pull request:
## Summary & Motivation

Explains how to set `owners` and `tags` on Component using `get_spec`.

## How I Tested These Changes

`yarn start`

## Changelog

NOCHANGELOG

cmpadden commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29043?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#29043** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29043?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/29043?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#28993** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/28993?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

schrockn reviewed (commented):
One question here is naming, in light of the fact that adding a `metadata` field inline with the rest of our definitions is a possibility. I also think we should be inline with our definition-level properties as much as possible (e.g. `owners` instead of `authors`)

cmpadden commented:
> One question here is naming, in light of the fact that adding a `metadata` field inline with the rest of our definitions is a possibility. I also think we should be inline with our definition-level properties as much as possible (e.g. `owners` instead of `authors`)

`owners` makes sense; though I'm now realizing that author and owners are different concepts. Dagster could be the author of a Component, and an organization could have several different owners for various implementations of that component.

Are you proposing that we move to top-level attributes instead of the `get_metadata` method, or a combination of both?

```
class MyComponent(Component):
    owners = []
    tags = []
    
    def get_metadata(self):
        return ComponentMetadata(self.owners, self.tags)
```

schrockn commented:
No I don't think they should be classvars as they are ambiguous to whether they apply to the class or the instance:

```python
class Foo:
    bar = 2


f = Foo()
print(f.bar)
print(Foo.bar)
```

prints 2 twice

`get_metadata` should be a class method I'm just questioning its name given that `metadata` could be a property of it one day.

`get_component_type_metadata` would be the long form. `get_component_spec` could be another.

cmpadden commented:
@schrockn I've updated the method to `get_component_type_metadata` and renamed `author` to `owners`. Allowing people to set multiple owners. Here's an example of how it's currently rendered:
![image](https://github.com/user-attachments/assets/5a9aebe0-8a3c-4fcd-b080-384eb3dc4b9e)
